### DISCUSSION TITLE
onChange issue with typescript

### DISCUSSION THREAD

**Discussion by teifler** [JedWatson/react-select#5264] [Q&A]
Hey fellow hackers,

I have the following issue I have a react select field and I want to add a onChange function but if I set the type to Props I get the following error message:

> TS2769: No overload matches this call.
>   Overload 1 of 2, '(props: { form?: string | undefined; name?: string | undefined; ref?: Ref<Select<unknown, boolean, GroupBase<unknown>>> | undefined; 'aria-errormessage'?: string | undefined; ... 67 more ...; defaultMenuIsOpen?: boolean | undefined; } & { ...; } & { ...; }): ReactElement<...>', gave the following error.
>     Type '(arg0: string[]) => void' is not assignable to type '(newValue: unknown, actionMeta: ActionMeta<unknown>) => void'.
>       Types of parameters 'arg0' and 'newValue' are incompatible.
>         Type 'unknown' is not assignable to type 'string[]'.
>   Overload 2 of 2, '(props: StyledComponentPropsWithAs<StateManagedSelect, any, {}, never, StateManagedSelect, StateManagedSelect>): ReactElement<...>', gave the following error.
>     Type '(arg0: string[]) => void' is not assignable to type '(newValue: unknown, actionMeta: ActionMeta<unknown>) => void'.

React Select Code
![image](https://user-images.githubusercontent.com/10044263/178103162-3117953c-b525-4a96-842b-bf91fd4004f3.png)

onChange function:
![image](https://user-images.githubusercontent.com/10044263/178103284-528fa093-128b-4fed-a04d-dd073b57f061.png)

Props Interface and options
![image](https://user-images.githubusercontent.com/10044263/178103187-4602c978-cb17-4a59-9c05-d22f82d0ad38.png)

If I use any in the onChange function it works but I want to use a type.. I tried to fix this issue since two hours any help is much appreciated!

Thanks,
Tim

**Comment by manjushsh** [comment]
can you share a Code sandbox link?

**Reply by teifler** [reply]
Hey Manjush,

I tried to get the project on codesandbox but I get weird errors.. Maybe anyone can help with that, is that related to the package json?
https://codesandbox.io/s/cool-ives-hf7mpr?file=/.eslintrc.js:497-551
